adapter/internal/oasparser/envoyconf/response_mapper.go

```diff
@@ -75,3 +107,107 @@ func genErrorResponseMapper(statusCode uint32, errorCode int32, message string,
 	}
 	return mapper
 }
+
+func genSoap12ErrorResponseMapper(flag string, statusCode uint32, errorCode int32, message string, description string) *hcmv3.ResponseMapper {
+	msg, _ := soaputils.GenerateSoapFaultMessage(soap12ProtocolVersion, message, description, strconv.Itoa(int(errorCode)))
+
+	mapper := &hcmv3.ResponseMapper{
```

Comment:
Since both the genSoap<version>ErrorResponseMappers returns the same response mapper except for the arrays inside the AndFilter, shall we use a common sub method to gen the &hcmv3.ResponseMapper for soap?

Reply:
since we only have two of them (soap 1.1 & 1.2) I don't think we need to add a common sub method here. Also if we add such, it might be little bit complicated as we need to have some if statements inside that method to switch between soap 1.1 & 1.2..

Reply:
what do you think of something like this? It would reduce the source code width as well?
```
func genSoap12ErrorResponseMapper(flag string, statusCode uint32, errorCode int32, message string, description string) *hcmv3.ResponseMapper {
	filters := []*access_logv3.AccessLogFilter{
		{
			FilterSpecifier: genExactMatchHeaderFilter(contentTypeHeaderName, contentTypeHeaderSoap),
		},
		{
			FilterSpecifier: genResponseFlagFilter(flag),
		},
	}

	return genSoapErrorResponseMapper(statusCode, errorCode, message, description, contentTypeHeaderSoap, filters)
}

func genSoap11ErrorResponseMapper(flag string, statusCode uint32, errorCode int32, message string, description string) *hcmv3.ResponseMapper {
	filters := []*access_logv3.AccessLogFilter{
		{
			FilterSpecifier: genExactMatchHeaderFilter(contentTypeHeaderName, contentTypeHeaderXML),
		},
		{
			FilterSpecifier: genPresentMatchHeaderFilter(soapActionHeaderName),
		},
		{
			FilterSpecifier: genResponseFlagFilter(flag),
		},
	}
	return genSoapErrorResponseMapper(statusCode, errorCode, message, description, contentTypeHeaderXML, filters)
}

func genSoapErrorResponseMapper(statusCode uint32, errorCode int32, message, description, contentTypeHeader string,
	filters []*access_logv3.AccessLogFilter) *hcmv3.ResponseMapper {

	msg, _ := soaputils.GenerateSoapFaultMessage(soap11ProtocolVersion, message, description, strconv.Itoa(int(errorCode)))

	mapper := &hcmv3.ResponseMapper{
		Filter: &access_logv3.AccessLogFilter{
			FilterSpecifier: &access_logv3.AccessLogFilter_AndFilter{
				AndFilter: &access_logv3.AndFilter{
					Filters: filters,
				},
			},
		},
		StatusCode: wrapperspb.UInt32(statusCode),
		BodyFormatOverride: &corev3.SubstitutionFormatString{
			Format: &corev3.SubstitutionFormatString_TextFormatSource{
				TextFormatSource: &corev3.DataSource{
					Specifier: &corev3.DataSource_InlineString{
						InlineString: msg,
					},
				},
			},
			ContentType: contentTypeHeader,
		},
	}
	return mapper
}

func genResponseFlagFilter(flag string) *access_logv3.AccessLogFilter_ResponseFlagFilter {
	return &access_logv3.AccessLogFilter_ResponseFlagFilter{
		ResponseFlagFilter: &access_logv3.ResponseFlagFilter{
			Flags: []string{flag},
		},
	}
}

func genPresentMatchHeaderFilter(headerName string) *access_logv3.AccessLogFilter_HeaderFilter {
	return &access_logv3.AccessLogFilter_HeaderFilter{
		HeaderFilter: &access_logv3.HeaderFilter{
			Header: &envoy_config_route_v3.HeaderMatcher{
				Name: soapActionHeaderName,
				HeaderMatchSpecifier: &envoy_config_route_v3.HeaderMatcher_PresentMatch{
					PresentMatch: true,
				},
			},
		},
	}
}

func genExactMatchHeaderFilter(headerName, headerValue string) *access_logv3.AccessLogFilter_HeaderFilter {
	return &access_logv3.AccessLogFilter_HeaderFilter{
		HeaderFilter: &access_logv3.HeaderFilter{
			Header: &envoy_config_route_v3.HeaderMatcher{
				Name: headerName,
				HeaderMatchSpecifier: &envoy_config_route_v3.HeaderMatcher_ExactMatch{
					ExactMatch: headerValue,
				},
			},
		},
	}
}
```

Reply:
Thanks for the suggestion, updated the changes with https://github.com/wso2/product-microgateway/pull/2972/commits/8b6acc3389596986055249ec65e5e0f7fa5a175c